### ISSUE TITLE
prefer_final_fields false positive with `++`

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/linter#612]
In SDK 1.23.0-dev.11.7

```
class Foo {
  int _foo = 0;
  
  int count() {
    return _foo++;
  }
}
```

```
INFO: Private field could be final. ([angular2] lib/test.dart:4)
```

**Comment by alexeieleusis** [comment]
This is already fixed, I wonder if you are seeing it because the SDK is shipped with a version that does not have the fix yet.

Tested here https://github.com/dart-lang/linter/blob/master/test/rules/prefer_final_fields.dart#L31

**Comment by kevmoo** [comment]
Thanks, @alexeieleusis – yup. Just need to get this synced.

**Comment by kevmoo** [comment]
It's been brought into the SDK here - https://github.com/dart-lang/sdk/commit/2bc70e6a263fd532390f0face0cbed0090052de0

But not in 1.23.0
